src/olympia/accounts/helpers.py

```diff
@@ -20,6 +47,42 @@ def fxa_config(context):
     return config
 
 
+def link_with_final_destination(context, base):
+    return urlparams(base, to=path_with_query(context['request']))
+
+
+def path_with_query(request):
+    next_path = request.path
+    qs = request.GET.urlencode()
+    if qs:
+        return '{next_path}?{qs}'.format(next_path=next_path, qs=qs)
```

Comment:
`request.path` is unicode, if it contains some utf-8 chars (like on some add-on or user pages, where we allow utf-8 chars in slugs) you'll be in trouble here, add `u` prefix.